app/controllers/ExternalServiceRedirectController.scala

```diff
@@ -35,16 +36,12 @@ class ExternalServiceRedirectController @Inject()(
 
   def auditInvalidateCacheAndRedirectService(serviceAndIFormName: String): Action[AnyContent] =
     (authenticate andThen validatePerson).async { implicit request =>
-      {
-
-        (for {
-          redirectUri <- auditService.sendAuditEventAndGetRedirectUri(request.taiUser.nino, serviceAndIFormName)
-          _           <- sessionService.invalidateCache()
-        } yield {
-          Redirect(redirectUri)
-        }) recover {
-          case _ => errorPagesHandler.internalServerError("Unable to audit and redirect")
-        }
+      (
+        auditService.sendAuditEventAndGetRedirectUri(request.taiUser.nino, serviceAndIFormName),
```

Comment:
Should this one be parallelised? If the audit event can't be send previously we would invalidate the cache. Now this behaviour won't happen, because no matter what we will always invalidate the Cache.

Reply:
Actually, after looking at the `sendAuditEventAndGetRedirectUri` method it can't fail as it always returns: 
`Future.successful(redirectUri)`

I think instead it should do: 
`sendIformRedirectUriAuditEvent(nino, fetchPath(request), auditEventName).map{ _ => redirectUri} `

And then the behaviour I explain would happen... Otherwise they both do the same thing :D

Reply:
@vivekhmrc whats your thoughts on this. Should we change this to use sendIformRedirectUriAuditEvent instead?

Reply:
@sergi-hmrc / @brendanstraughan Morning guys as a thumb rule the audit event is always in the end to capture the completion of the operations (success or failure). 

I think the logic for auditInvalidateCacheAndRedirectService should have been that the cache is dropped first and then the audit event written. Probably that's the reason why `sendAuditEventAndGetRedirectUri` is always returning `Future.successful(redirectUri)` regardless of the outcome of [auditConnector.sendEvent](url) `createAndSendAuditEvent`

Reply:
Cool. In that case, the code should be sequential: 
```
for {
          _           <- sessionService.invalidateCache()
          redirectUri <- auditService.sendAuditEventAndGetRedirectUri(request.taiUser.nino, serviceAndIFormName)
        } yield Redirect(redirectUri)
```

Finally the Future.successful is more likely to be an error of misunderstanding how futures work than anything else (after seeing some other code in tai...). Should we map the redirect on the result so it's sequential rather than fire and forget?